### PR TITLE
Factor out _write() implementation

### PR DESCRIPTION
**Pull request by marbre**
Co-authored-by: Lucas Camphausen <lucas.camphausen@iml.fraunhofer.de>

### CODE REVIEW

**Review by lucas-camp — COMMENTED** [review]
Looks good. We can further remove the include of `unistd.h` in some files.

File: utils/nrf52840_cmsis.c [hunk 1]
```diff
@@ -5,7 +5,6 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-#include <errno.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <unistd.h>
```
Comment (lucas-camp) [inline utils/nrf52840_cmsis.c:10]:
```suggestion
```

File: utils/stm32f4_cmsis.c [hunk 2]
```diff
@@ -5,7 +5,6 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-#include <errno.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <unistd.h>
```
Comment (lucas-camp) [inline utils/stm32f4_cmsis.c:10]:
```suggestion
```

File: utils/stm32f7_cmsis.c [hunk 3]
```diff
@@ -5,7 +5,6 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-#include <errno.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <unistd.h>
```
Comment (lucas-camp) [inline utils/stm32f7_cmsis.c:10]:
```suggestion
```

**Comment by marbre** [comment]
> Looks good. We can further remove the include of `unistd.h` in some files.

Good point. Done.

**Review by lucas-camp — APPROVED** [review]

**Pull request merged**